### DISCUSSION TITLE
Cannot rebind the Return key in major mode calling gptel-menu

### DISCUSSION THREAD

**Discussion by scaramouche1** [karthink/gptel#247] [General]
The key `Return` key ceases to work in `gptel-menu` if it is redefined in the major mode calling `gptel-menu`.  For example, if in the configuration of `org-mode`, I set:

    (define-key org-mode-map (kbd "<return>") 'newline-and-indent)

Then, when called from `org-mode`, pressing the `Return` key in `gptel-menu` results in the following error:

    Unbound suffix: '<return>' (use 'C-g' to abort, '?' for help) [newline-and-indent]

How can I redefine `Return` and still use it from within the transient menu?

(I'm also not understanding why this problem happens, as the transient menu should check which key is being pressed, not what function that key calls in the major mode).

**Comment by karthink** [comment]
I'm not sure why it happens.  I've rebound `<return>` in Org mode too but can't reproduce this.  See also #75 and #182.

> How can I redefine Return and still use it from within the transient menu?

See the [FAQ](https://github.com/karthink/gptel?tab=readme-ov-file#doom-emacs-sending-a-query-from-the-gptel-menu-fails-because-of-a-key-conflict-with-org-mode).
